spec/system/admin/bulk_url_upload_spec.rb

```diff
@@ -1,27 +1,38 @@
 # frozen_string_literal: true
 
+shared_examples 'a failed bulk upload with error' do |error_message|
+  it 'sends us back to the bulk upload page' do
+    do_bulk_upload
+    expect(page).to have_text('Bulk Search.gov URL Upload')
+  end
+
+  it 'shows an error message' do
+    do_bulk_upload
+    expect(page).to have_text(error_message)
+  end
+end
+
 describe 'Bulk URL upload' do
   include ActiveJob::TestHelper
 
-  subject(:bulk_upload) do
+  subject(:do_bulk_upload) do
     perform_enqueued_jobs do
       visit url
-      attach_file('bulk_upload_urls', url_file)
+      attach_file('bulk_upload_urls', upload_file)
       click_button('Upload')
     end
   end
 
   let(:url) { '/admin/bulk_url_upload' }
-  let(:url_filedir) { 'txt' }
-  let(:url_filename) { 'good_url_file.txt' }
-  let(:url_file) { file_fixture("#{url_filedir}/#{url_filename}") }
-  let(:urls) { File.open(url_file).readlines.map(&:strip) }
+  let(:upload_file) { file_fixture("txt/#{upload_filename}") }
+  let(:urls) { File.open(upload_file, 'r:bom|utf-8').readlines.map(&:strip) }
   let(:searchgov_domains) do
     urls.reduce(Set.new) do |searchgov_domains, raw_url|
       parsed_url = URI(raw_url)
       domain = parsed_url.host
       searchgov_domain = SearchgovDomain.find_by(domain: domain)
-      searchgov_domains << searchgov_domain if searchgov_domain
+      searchgov_domains = searchgov_domains << searchgov_domain if searchgov_domain
+      searchgov_domains
     end
   end
```

Comment:
This is also a lot more code than we need to test the reindexing. My recommendation to clean this up:
```ruby
  let(:urls) { File.open(upload_file).readlines.map(&:strip) }
  let(:searchgov_domain) { searchgov_domains(:agency_gov) }

  before { allow(searchgov_domain).to receive(:index_urls) }
  
  ...
    describe 'bulk uploading a file of URLs' do
    ...
      it 're-indexes the domains for the URLs' do
        reindexed_domains = Set.new
        allow_any_instance_of(SearchgovDomain).to receive(:index_urls) do |searchgov_domain|
          reindexed_domains << searchgov_domain
        end
        do_bulk_upload
        expect(reindexed_domains).to eq [searchgov_domain]
      end
```
I also discovered that there are two fixture files for SearchgovDomains: `searchgov_domains.yml` and `searchgov_domain.yml`. Please move any necessary fixtures to `spec/fixtures/searchgov_domains.yml` and delete `spec/fixtures/searchgov_domain.yml`.

Reply:
Cleaned up.